main.go

```diff
@@ -81,31 +81,43 @@ Generate an identity seed and launch a gateway:
 			Name:    "seed",
 			Value:   "",
 			EnvVars: []string{"RAINBOW_SEED"},
-			Usage:   "Seed to derive peerID from. Generate with gen-seed. Needs --seed-index. Best to use $CREDENTIALS_DIRECTORY/seed or $RAINBOW_DATADIR/seed.",
+			Usage:   "Seed to derive peerID from. Generate with gen-seed. Needs --seed-index. Best to use $CREDENTIALS_DIRECTORY/seed or $RAINBOW_DATADIR/seed",
 		},
 		&cli.IntFlag{
 			Name:    "seed-index",
 			Value:   -1,
 			EnvVars: []string{"RAINBOW_SEED_INDEX"},
 			Usage:   "Index to derivate the peerID (needs --seed)",
 		},
-		&cli.StringFlag{
+		&cli.BoolFlag{
+			Name:    "seeds-peering",
+			Value:   false,
+			EnvVars: []string{"RAINBOW_SEEDS_PEERING"},
+			Usage:   "Automatic peering with peers with the same seed (requires --seed and --seed-index). Automatically enables --dht-routing and --dht-shared-host",
```

Comment:
:thought_balloon:  Having seed-based peering and cache sharing at the cost of enabling DHT routing and switching to shared host feels like one step forward, two steps back. 

What are our options for avoiding running Amino DHT all the time? 
We only need it for multiaddr discovery, right?

- Run someguy with Delegated PUTS via /routing/v1 enabled?
- Manually setting multiaddrs of other seeds via  `RAINBOW_PEERING`?

Reply:
FYSA @aschmahmann raised additional complexities related to holepunching and DHT-based discovery  ([thread for @hacdias](https://ipshipyard.slack.com/archives/C06TGA7P3B2/p1712332362550569)) and we've decided to adjust the scope/priorities.

- To unblock cache-sharing (#109), we want to reduce complexity. If we remove dependency on peer discovery and routing, that removes the requirement of enabling DHT for cache-sharing. 
- For now, to enable cache sharing on ipfs.io gateway, we will precompute multiaddrs from seed+index+ip|dns in scripts that deploy our infra and then pass as peering safelist to rainbows via `RAINBOW_PEERING=/dns/n.example.net/4001/quic-v1/p2p/seed-peerid@n`. 

This allows us to enable cache sharing with manually peered rainbow nodes.
The seed-based automatic peering and multiaddr discovery is still something we want, but has lower priority.

@hacdias I think the next step here is to decouple these two features:
- park seed-based peering (`RAINBOW_SEED_PEERING` and related code/configuration) in this PR for now (keep as draft)
  - Adin will be landing some fixes to unblock automatic peering (e.g. https://github.com/ipfs/rainbow/pull/112)
- open a new PR focused on cache sharing, 
  - make it use multiaddrs and peerids from  `RAINBOW_PEERING` as safelist for something like `RAINBOW_CACHE_SHARING=off|peering-only|everyone` (where `off` is default, and `everyone` is something we can use during testing)
  - :thought_balloon: when parsing `RAINBOW_PEERING`  replace `/p2p/rainbow-seed-peerid@n`  with generated peerid based on `n` as seed index
    - `/dns/42.example.net/4001/quic-v1/p2p/rainbow-seed-peerid@42` → `/dns/42.example.net/4001/quic-v1/p2p/12D..`

Reply:
> Adin will be landing some fixes to unblock automatic peering

To clarify (and repeat for those who don't have access to the linked thread ...sorry my fault):
- #112 will do the basics of holepunching (i.e. where the client is publicly reachable but wants to reach nodes behind NATs) for nodes listening on public addresses
- However, for both the seeds feature and the holepunching feature to fully work some amount of Amino DHT usage is likely required (although it's slightly different for each). The plan is to push this to a separate PR where we can figure out how to do this (e.g. can/should we run a lighter DHT client and can/should we make changes in the boxo/bitswap client to limit adding peers to the bitswap client's tracking to not include those that we're only bothering to contact for unrelated reasons like advertising our multiaddrs or discovering our public addresses)